apps/src/accounts/ChangeEmail/ChangeEmailForm.jsx

```diff
@@ -0,0 +1,133 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import FontAwesomeV6Icon from '@cdo/apps/componentLibrary/fontAwesomeV6Icon/FontAwesomeV6Icon';
+import Link from '@cdo/apps/componentLibrary/link/Link';
+import {RadioButtonsGroup} from '@cdo/apps/componentLibrary/radioButton';
+import TextField from '@cdo/apps/componentLibrary/textField/TextField';
+import Typography from '@cdo/apps/componentLibrary/typography/Typography';
+import i18n from '@cdo/locale';
+
+import {pegasus} from '../../lib/util/urlHelpers';
+
+import styles from './formStyle.module.scss';
+
+export default class ChangeEmailForm extends React.Component {
+  static propTypes = {
+    values: PropTypes.shape({
+      newEmail: PropTypes.string,
+      currentPassword: PropTypes.string,
+      emailOptIn: PropTypes.string,
+    }).isRequired,
+    validationErrors: PropTypes.shape({
+      newEmail: PropTypes.string,
+      currentPassword: PropTypes.string,
+      emailOptIn: PropTypes.string,
+    }).isRequired,
+    userType: PropTypes.oneOf(['teacher', 'student']).isRequired,
+    isPasswordRequired: PropTypes.bool.isRequired,
+    disabled: PropTypes.bool,
+    onChange: PropTypes.func.isRequired,
+  };
+
+  state = {touched: false};
```

Comment:
Is there a reason for using `state` here rather than `useState` like the other files in this PR?

Reply:
ah yes, I didn't refactor this to a react functional component. it's still a class component so the state property looks like this, and it's manipulated with `this.setState`.

dang, because I moved the file it looks like I wrote it from scratch, but I swear I only made small changes 😞

Reply:
you know what? I should leave it better than I found it. I'll refactor to use a functional component

Reply:
I refactored ChangeEmailForm to a functional component with types, but hit a wall when converting ChangeEmailModal. Reason being the class component is tested using enzyme, and it uses some unfortunate methods like `wrapper.setState(...)` which you can't do in a functional component. since the modal doesn't use any of the deprecated class methods like `componentDidMount` it's safe to keep it the old way for the time being, imho